realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -90,16 +74,32 @@ public void realmClosed(RealmConfiguration configuration) {
     public Object[] getSyncConfigurationOptions(RealmConfiguration config) {
         if (config instanceof SyncConfiguration) {
             SyncConfiguration syncConfig = (SyncConfiguration) config;
-            SyncUser user = syncConfig.getUser();
+            RealmUser user = syncConfig.getUser();
+            RealmApp app = user.getApp();
             String rosServerUrl = syncConfig.getServerUrl().toString();
-            String rosUserIdentity = user.getIdentity();
-            String syncRealmAuthUrl = user.getAuthenticationUrl().toString();
-            String syncUserRefreshToken = user.getRefreshToken().toJson().toString();
-            String syncUserAccessToken = user.getAccessToken(((SyncConfiguration) config)).toJson().toString();
+            String rosUserIdentity = user.getId();
+            String syncRealmAuthUrl = user.getApp().getConfiguration().getBaseUrl().toString();
+            String syncUserRefreshToken = user.getRefreshToken();
+            String syncUserAccessToken = user.getAccessToken();
             byte sessionStopPolicy = syncConfig.getSessionStopPolicy().getNativeValue();
             String urlPrefix = syncConfig.getUrlPrefix();
-            String customAuthorizationHeaderName = SyncManager.getAuthorizationHeaderName(syncConfig.getServerUrl());
-            Map<String, String> customHeaders = SyncManager.getCustomRequestHeaders(syncConfig.getServerUrl());
+            String customAuthorizationHeaderName = app.getConfiguration().getAuthorizationHeaderName();
+            Map<String, String> customHeaders = app.getConfiguration().getCustomRequestHeaders();
+
+            // Temporary work-around for serializing supported bson values
+            BsonValue val = syncConfig.getPartitionValue();
+            String partitionValue = null;
+            if (val.isString()) {
```

Comment:
Unfortunately I still have not found a nice way to serialize to json without adding json keys using org.bson-library